integrations/qdrant/src/haystack_integrations/document_stores/qdrant/converters.py

```diff
@@ -58,13 +65,38 @@ def convert_id(self, _id: str) -> str:
 
 
 class QdrantToHaystack:
-    def __init__(self, content_field: str, name_field: str, embedding_field: str):
+    def __init__(
+        self,
+        content_field: str,
+        name_field: str,
+        embedding_field: str,
+        use_sparse_embeddings: bool,  # noqa: FBT001
+        sparse_embedding_field: str,
+    ):
         self.content_field = content_field
         self.name_field = name_field
         self.embedding_field = embedding_field
+        self.use_sparse_embeddings = use_sparse_embeddings
+        self.sparse_embedding_field = sparse_embedding_field
 
     def point_to_document(self, point: QdrantPoint) -> Document:
         payload = {**point.payload}
-        payload["embedding"] = point.vector if hasattr(point, "vector") else None
         payload["score"] = point.score if hasattr(point, "score") else None
+        if not self.use_sparse_embeddings:
+            payload["embedding"] = point.vector if hasattr(point, "vector") else None
+
+        if self.use_sparse_embeddings:
+            if hasattr(point, "vector") and point.vector is not None and DENSE_VECTORS_NAME in point.vector:
+                payload["embedding"] = point.vector[DENSE_VECTORS_NAME]
+            else:
+                payload["embedding"] = None
+
+            if hasattr(point, "vector") and point.vector is not None and SPARSE_VECTORS_NAME in point.vector:
+                parse_vector_dict = {
+                    "indices": point.vector[SPARSE_VECTORS_NAME].indices,
+                    "values": point.vector[SPARSE_VECTORS_NAME].values,
+                }
+                payload["sparse_embedding"] = parse_vector_dict
+            else:
+                payload["sparse_embedding"] = None
```

Comment:
Handle previous collection without named dense vector (and without sparse) so that there is not breaking change.